### PR TITLE
format listenning address correctly for IPv6 ip address

### PR DESCRIPTION
**Pull request by ahrtr**
Fix [issues/13548 ](https://github.com/etcd-io/etcd/issues/13548 )

We should add more test for IPv6, but probably it would be better to resolve the test in a separate PR. 

@xiang90 @ptabor @serathius

### CODE REVIEW

File: server/proxy/tcpproxy/userspace.go [hunk 1]
```diff
@@ -69,13 +69,31 @@ type TCPProxy struct {
 	pickCount int // for round robin
 }
 
+// The parameter host is returned by net.SplitHostPort previously,
+// so it must be a valid host. This function is only to check whether
+// it's an IPv6 IP address.
+func isIPv6(host string) bool {
+	for i := 0; i < len(host); i++ {
```
Comment (serathius) [inline server/proxy/tcpproxy/userspace.go:76]: This code just looks for ":" in string. Why not use strings.Index(host, ":") != -1

Comment (ahrtr) [reply server/proxy/tcpproxy/userspace.go]: Good point. String.IndexByte is better. Resolved.

**Comment by ahrtr** [comment]
Just added a unit test case, PTAL, thanks.

File: server/proxy/tcpproxy/userspace.go [hunk 2]
```diff
@@ -69,13 +70,29 @@ type TCPProxy struct {
 	pickCount int // for round robin
 }
 
+// The parameter host is returned by net.SplitHostPort previously,
+// so it must be a valid host. This function is only to check whether
+// it's an IPv6 IP address.
+func isIPv6(host string) bool {
+	return strings.IndexByte(host, byte(':')) >= 0
```
Comment (serathius) [inline server/proxy/tcpproxy/userspace.go:77]:
```suggestion
	return strings.IndexRune(host, ':') != -1
```

Comment (ahrtr) [reply server/proxy/tcpproxy/userspace.go]: Thanks. I agree that IndexRune is better. Based on the golang spec: `A rune literal represents a rune constant, an integer value identifying a Unicode code point`

Fixed.

**Review by serathius — APPROVED** [review]
Thanks!

**Review by ptabor — APPROVED** [review]
Thank you.

**Comment by ahrtr** [comment]
Just rebased this PR and added a CHANGELOG-3.6 item.

**Review by spzala — APPROVED** [review]
lgmt, thanks @ahrtr

**Pull request merged**